python/oneflow/nn/modules/global_cast.py

```diff
@@ -75,16 +77,56 @@ def global_to_global_op(
 
 
 def to_global_op(input, placement=None, sbp=None, **kwargs):
-    assert isinstance(input, Tensor)
-
-    if input.is_global:
-        return global_to_global_op(input=input, placement=placement, sbp=sbp, **kwargs)
-    else:
-        if "grad_sbp" in kwargs:
-            del kwargs["grad_sbp"]
-        return local_to_global_op(input=input, placement=placement, sbp=sbp, **kwargs)
+    r"""Converts the input to global if it is local, otherwise performs designated placement and/or sbp conversion.
+    
+    It performs the same conversion as :func:`oneflow.Tensor.to_global` to the tensor or every tensor in the state dict.
+
+    Note:
+        Both placement and sbp are required if the input is local, otherwise at least one of placement and sbp is required.
+
+    Args:
+        input (flow.Tensor/None/list/tuple/dict): the input that needs to be converted.
+        placement (flow.placement, optional):  the desired placement of the input. Default: None
+        sbp (flow.sbp.sbp or list/tuple of flow.sbp.sbp, optional): the desired sbp of the input. Default: None
+    
+    Returns:
+        The converted tensor / state dict.
+    """
+    input_tree = IONode(value=input)
+
+    def leaf_fn(node):
+        if isinstance(node._value, Tensor):
```

Comment:
Define a `to_global_tensor` function to do this.

if is Tensor, call `to_global_tensor`, else build a `IONode`

Reply:
Corrected